i18n/middleware.py

```diff
@@ -0,0 +1,36 @@
+import re
+import warnings
+
+from django.conf import settings
+from django.utils import translation
+from django.http import HttpResponseRedirect
+from django.middleware.locale import LocaleMiddleware
+
+def get_language_from_path_strict(path, allowed_languages):
+    language_code_prefix_re = re.compile(r'^/([\w-]+)(/|$)')
```

Comment:
is it necessary to use regex for this? I figure we could just do something like

```python
def get_language_from_path_strict(path, allowed_languages):
  for language_code in allowed_languages:
    if path.startswith('/{}/'.format(language_code)):
      return language_code

  return settings.LANGUAGE_CODE
```

Reply:
Changed